features/networking/operator.feature

```diff
@@ -13,3 +13,167 @@ Feature: Operator related networking scenarios
     Then the expression should be true> cb.operator_status["status"]=="True"
     #Confirm whether network operator version matches with ocp version
     And the expression should be true> cluster_operator('network').version_exists?(version: cb.ocp_version)
+
+
+  # @author bmeng@redhat.com
+  # @case_id OCP-22201
+  @admin
+  Scenario: Should have a clusteroperator object created under config.openshift.io api group for network-operator
+    Given the master version > "3.11"
+    # Check the operator object has version
+    Given the expression should be true> cluster_operator('network').versions.length > 0
+    # Check the operator object has status for Failing|Progressing|Available
+    And the expression should be true> cluster_operator('network').condition(type: 'Available')['status'] == "True"
+    And the expression should be true> cluster_operator('network').condition(type: 'Failing')['status'] == "False"
+    And the expression should be true> cluster_operator('network').condition(type: 'Progressing')['status'] == "False"
+
+
+  # @author bmeng@redhat.com
+  # @case_id OCP-22419
+  @admin
+  @destructive
+  Scenario: The clusteroperator should be able to reflect the realtime status of the network when the config has problem
+    Given the master version > "3.11"
+    # Check that the operator is not failing
+    Given the expression should be true> cluster_operator('network').condition(type: 'Failing')['status'] == "False"
+    # Copy the value of the networktype for backup
+    When I run the :get admin command with:
+      | resource      | network.config.openshift.io |
+      | resource_name | cluster                     |
+      | template      | {{.spec.networkType}}       |
+    Then the step should succeed
+    And evaluation of `@result[:response]` is stored in the :network_type clipboard
+    # Do some modification on the network.config.openshift.io
+    When I run the :patch admin command with:
+      | resource      | network.config.openshift.io     |
+      | resource_name | cluster                         |
+      | p             | {"spec":{"networkType":"None"}} |
+      | type          | merge                           |
+    Then the step should succeed
+    Given I register clean-up steps:
+    """
+    When I run the :patch admin command with:
+      | resource      | network.config.openshift.io                       |
+      | resource_name | cluster                                           |
+      | p             | {"spec":{"networkType":"<%= cb.network_type %>"}} |
+      | type          | merge                                             |
+    Then the step should succeed
+    """
```

Comment:
worth writing a step `cluster network type is restored after scenario`

Reply:
Actually, we do not support changing the networkType, this testing is just checking that the operator will report when problem. So I think we will not have many cases to use this method.